content/academy/puppeteer_playwright/browser_contexts.md

```diff
@@ -0,0 +1,134 @@
+// Open a page on the newly created iPhone context
+const iPhonePage = await iPhoneContext.newPage();
+
+const android = devices['Galaxy Note 3'];
+// Create a new context for our Android emulation
+const androidContext = await browser.newContext({ ...android });
+// Open a page on the newly created Android context
+const androidPage = await androidContext.newPage();
+
+// The code in the next step will go here
+
+await browser.close();
+</marked-tab>
+<marked-tab header="Puppeteer" lang="javascript">
+import puppeteer from 'puppeteer';
+
+// Launch the browser
+const browser = await puppeteer.launch({ headless: false });
+
+const iPhone = puppeteer.devices['iPhone 11 Pro'];
+// Create a new context for our iPhone emulation
+const iPhoneContext = await browser.createIncognitoBrowserContext();
+// Open a page on the newly created iPhone context
+const iPhonePage = await iPhoneContext.newPage();
+// Emulate the device
+await iPhonePage.emulate(iPhone);
+
+const android = puppeteer.devices['Galaxy Note 3'];
+// Create a new context for our Android emulation
+const androidContext = await browser.createIncognitoBrowserContext();
+// Open a page on the newly created Android context
+const androidPage = await androidContext.newPage();
+// Emulate the device
+await androidPage.emulate(android);
+
+// The code in the next step will go here
+
+await browser.close();
+</marked-tab>
+```
+
+Then, we'll make both `iPhonePage` and `androidPage` visit [deviceinfo.me](https://www.deviceinfo.me/), which is a website that  displays the type of device you have, the operating system you're using, and more device and location-specific information.
+
+```JavaScript
+// Go to deviceinfo.me on both at the same time
+await Promise.all([iPhonePage.goto('https://www.deviceinfo.me/'), androidPage.goto('https://www.deviceinfo.me/')]);
+
+// Wait for 10 seconds on both before shutting down
+await Promise.all([iPhonePage.waitForTimeout(10000), androidPage.waitForTimeout(10000)]);
+```
+
+Let's go ahead and run our code and analyze the data on each **deviceinfo.me** page. Here's what we see:
+
+![deviceinfo.me results for both browser contexts]({{@asset puppeteer_playwright/images/dual-contexts.jpg}})
+
+We see that **deviceinfo.me** detects both contexts as using different devices, despite the fact they're visiting the same page at the same time.  This shows firsthand that different browser contexts can have totally different configurations, as they all have separate sessions.
+
+## [](#accessing-browser-contexts) Accessing browser contexts
+
+When working with multiple browser contexts, it can be difficult to keep track of all of them and repetitive when making changes to all of them. This is why the **Browser** instance returned from the `launch()` function also has a `contexts()` function (`newContexts()` in Puppeteer). This function returns an array of all the contexts that are currently attached to the browser.
```

Comment:
```suggestion
When working with multiple browser contexts, it can be difficult to keep track of all of them and repetitive when making changes to all of them. This is why the **Browser** instance returned from the `launch()` function also has a `contexts()` function (`browserContexts()` in Puppeteer). This function returns an array of all the contexts that are currently attached to the browser.
```